cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -59,3 +67,28 @@ func Run(gadgetRoot, device string, options *Options) error {
 
 	return nil
 }
+
+func ensureLayoutCompatibility(gadgetLayout, diskLayout *gadget.LaidOutVolume) error {
+	eq := func(ds, gs gadget.LaidOutStructure) bool {
+		dv := ds.VolumeStructure
+		gv := gs.VolumeStructure
+		return dv.Name == gv.Name && dv.Role == gv.Role && ds.StartOffset == gs.StartOffset && dv.Size == gv.Size && dv.Filesystem == gv.Filesystem
+	}
+	contains := func(haystack []gadget.LaidOutStructure, needle gadget.LaidOutStructure) bool {
+		for _, h := range haystack {
+			if eq(needle, h) {
+				return true
+			}
+		}
+		return false
+	}
+
+	// Check if all existing device partitions are also in gadget
+	for _, ds := range diskLayout.LaidOutStructure {
+		if !contains(gadgetLayout.LaidOutStructure, ds) {
+			return fmt.Errorf("cannot find disk partition %q (starting at %d) in gadget", ds.VolumeStructure.Label, ds.StartOffset)
```

Comment:
What if there is no filesystem label?

Maybe we could use:
```
fmt.Errorf("cannot find disk partition %s in gadget", ds.VerboseString())
```
Provided that `sfdisk.Layout()` properly populates LaidOutStructure.Index, we could then have a helper:
```
func (l LaidOutStructure) VerboseString() string {
    name := l.Name
    if name == "" {
       name = "<no-name>"
    }
    fs := ""
    if l.HasFilesystem() {
    fs = fmt.Sprintf(" fs:%s", l.Filesystem)
        if l.Label != "" {
            fs += fmt.Sprintf("{%q}", l.Label)
        }
    }
    return fmt.Sprintf("#%v (%q@%#x{%v}%s)", l.Index, name, l.StartOffset, l.Size, fs)
}
```
This should give something like
```
cannot find disk partition #1 ("foo"@0x123{123M} fs:ext4{"writable"}) in gadget
```